pallets/asset-manager/src/lib.rs

```diff
@@ -396,6 +428,54 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> Contains<MultiLocation> for Pallet<T> {
+		fn contains(location: &MultiLocation) -> bool {
+			let MultiLocation { parents, interior } = location;
+
+			match interior {
+				// this branch means sending tokens to relaychain.
```

Comment:
Shouldn't we check that parents is 1 for the relay chain ? Someone could be sending to the same parachain and then destination would be {parents: 0, X1(AccountId32)}